src/main/java/org/opensearch/security/user/UserService.java

```diff
@@ -0,0 +1,311 @@
+        this.client = client;
+    }
+
+    /**
+     * Load data for a given CType
+     * @param config CType whose data is to be loaded in-memory
+     * @return configuration loaded with given CType data
+     */
+    protected static final SecurityDynamicConfiguration<?> load(final CType config, boolean logComplianceEvent) {
+        SecurityDynamicConfiguration<?> loaded = configurationRepository.getConfigurationsFromIndex(Collections.singleton(config), logComplianceEvent).get(config).deepClone();
+        return DynamicConfigFactory.addStatics(loaded);
+    }
+
+    protected void saveAndUpdateConfiguration(final Client client, final CType cType,
+                                              final SecurityDynamicConfiguration<?> configuration) {
+        final IndexRequest ir = new IndexRequest(this.securityIndex);
+
+        // final String type = "_doc";
+        final String id = cType.toLCString();
+
+        configuration.removeStatic();
+
+        try {
+            client.index(ir.id(id)
+                            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+                            .setIfSeqNo(configuration.getSeqNo())
+                            .setIfPrimaryTerm(configuration.getPrimaryTerm())
+                            .source(id, XContentHelper.toXContent(configuration, XContentType.JSON, false)));
+        } catch (IOException e) {
+            throw ExceptionsHelper.convertToOpenSearchException(e);
+        }
+    }
+
+    /**
+     * This function will handle the creation or update of a user account.
+     *
+     * @param accountDetailsAsString A string JSON of different account configurations.
+     * @return
+     * @throws ServiceAccountRegistrationException
+     * @throws IOException
+     */
+    public SecurityDynamicConfiguration<?> createOrUpdateAccount(String accountDetailsAsString) throws IOException, AccountCreateOrUpdateException, ServiceAccountRegistrationException {
+
+
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode accountDetails = mapper.readTree(accountDetailsAsString);
+        final ObjectNode contentAsNode = (ObjectNode) accountDetails;
+        final SecurityJsonNode securityJsonNode = new SecurityJsonNode(contentAsNode);
+        final List<String> securityRoles = securityJsonNode.get("opendistro_security_roles").asList();
+        final SecurityDynamicConfiguration<?> internalUsersConfiguration = load(getConfigName(), false);
+
+        String accountName = securityJsonNode.get("name").asString();
+
+        if (accountName == null || accountName.length() == 0) { // Fail if field is present but empty
+            throw new AccountCreateOrUpdateException(NO_ACCOUNT_NAME_MESSAGE);
+        }
+
+        final Map<String, String> accountAttributes = new HashMap<>();
+
+        if (!securityJsonNode.get("service").isNull() && securityJsonNode.get("service").asString() == "true") { // If this is a service account
```

Comment:
Its difficult to see the schema changes that are being added for service accounts as is.  It might be cleaner to separate the service account updates into follow up PR - what do you think?

Reply:
I would prefer to leave this here since I am splitting it into methods. The more PRs the harder it is to keep velocity up. I will refactor to make it clearer.